examples/vision/image_enhancer.py

```diff
@@ -0,0 +1,297 @@
+fig.text(x=0.45, y=0.78, s="Dataset Visualization", fontweight="bold")
+fig.set_tight_layout(None)
+
+"""
+## Model
+
+We have used Pre-trained ```ResNet50V2``` U-Net as Encoder, The Feature Maps are
+extracted at different levels forming Downscaling path, Then ```Sub-Pixel Layers``` on
+the Upscaling path with skip-connections.
+"""
+
+
+def conv_unit(x, filters=64, padding="same"):
+    x = keras.layers.Conv2D(filters=filters, kernel_size=3, padding=padding)(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.Activation("relu")(x)
+    return x
+
+
+def subpixel_upsample(x, target_channels, upscale_ratio):
+    estimated_filters = target_channels * (upscale_ratio ** 2)
+    x = keras.layers.Conv2D(estimated_filters, 3, padding="same")(x)
+    x = keras.layers.BatchNormalization()(x)
+    x = keras.layers.LeakyReLU(0.2)(x)
+    x = tf.nn.depth_to_space(x, block_size=upscale_ratio)
+    return x
+
+
+def upscale_unit(previous_input, current_input, target_channels=3, upscale_ratio=2):
+    previous_low_scale = subpixel_upsample(
+        previous_input, target_channels=target_channels, upscale_ratio=upscale_ratio
+    )
+    current_high_scale = conv_unit(current_input, filters=target_channels)
+    concat_scales = tf.concat((previous_low_scale, current_high_scale), axis=-1)
+    feature_manifold = conv_unit(concat_scales, filters=target_channels)
+    return feature_manifold
+
+
+def denoiser():
+    base_model = ResNet50V2(
+        include_top=False, weights="imagenet", input_shape=(352, 608, 3)
+    )
+    encode0 = base_model.get_layer("input_1").output
+    encode1 = base_model.get_layer("conv1_conv").output
+    encode2 = base_model.get_layer("conv2_block3_1_relu").output
+    encode3 = base_model.get_layer("conv3_block4_1_relu").output
+
+    decode1 = upscale_unit(
+        previous_input=encode3, current_input=encode2, target_channels=128
+    )
+    decode2 = upscale_unit(
+        previous_input=decode1, current_input=encode1, target_channels=64
+    )
+    decode3 = upscale_unit(
+        previous_input=decode2, current_input=encode0, target_channels=32
+    )
+
+    out_conv = keras.layers.Conv2D(3, 5, padding="same", activation="sigmoid")(decode3)
+    denoiser_model = keras.Model(base_model.input, out_conv, name="denoiser")
+    denoiser_model.summary()
+    return denoiser_model
+
+
+"""
+## Training
+
+The Model is Trained with an Objective of minimizing ```Mean Squared Error```.
+"""
+
+denoiser_net = denoiser()
+
+train_step = len(train_source) // batch_size
+val_step = len(val_source) // batch_size
+early_stop = callbacks.EarlyStopping(patience=8)
+ckpt = callbacks.ModelCheckpoint(
+    "best_ckpt.h5", save_best_only=True, save_weights_only=True, verbose=1
+)
+
+optimizer = tf.keras.optimizers.Adam(1e-4)
+denoiser_net.compile(optimizer=optimizer, loss="mse", metrics=["mae"])
+
+denoiser_net.fit(
+    train_pipe,
+    validation_data=val_pipe,
+    batch_size=batch_size,
+    epochs=28,
```

Comment:
@anish9 make a variable for epoch beforehand, for ex.

```python
EPOCHS = 20
BATCH_SIZE = 64

model.fit( ..., epochs=EPOCHS, ...)
``` 

Also, as `keras` is imported from `tensorflow`, it would be nice to use 

```python
keras.optimizers.Adam(1e-4)

# instead of 
tf.keras.optimizers.Adam(1e-4)
```

And there are other places where such practice can be done, for example: `keras.utils.array_to_img` instead of `tf.keras.utils..`.